diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 31db4b3..835bc75 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -6,11 +6,10 @@ option(SCODES_DEBUG OFF)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTORCC ON)
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-find_package(QT NAMES Qt5 Qt6)
-find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Core Multimedia Concurrent Quick REQUIRED)
+find_package(Qt6 REQUIRED COMPONENTS Core Multimedia Concurrent Quick CorePrivate)
 
 set(COMMON_SOURCES
     ${CMAKE_CURRENT_SOURCE_DIR}/SBarcodeDecoder.cpp
@@ -282,29 +281,14 @@ set(COMMON_HEADERS
     ${CMAKE_CURRENT_SOURCE_DIR}/zxing-cpp/core/src/qrcode/QRWriter.h
 )
 
-if (QT_VERSION_MAJOR EQUAL 6)
-    set(SRC_FILES ${COMMON_SOURCES}
-        ${CMAKE_CURRENT_SOURCE_DIR}/SBarcodeScanner.cpp
-    )
-    set(HEADER_FILES ${COMMON_HEADERS}
-        ${CMAKE_CURRENT_SOURCE_DIR}/SBarcodeScanner.h
-    )
-	if(ANDROID)
-        find_package(Qt6 COMPONENTS GuiPrivate REQUIRED)
-    endif()
-endif()
-
-if (QT_VERSION_MAJOR EQUAL 5)
-    set(SRC_FILES ${COMMON_SOURCES}
-        ${CMAKE_CURRENT_SOURCE_DIR}/SBarcodeFilter.cpp
-    )
-    set(HEADER_FILES ${COMMON_HEADERS}
-        ${CMAKE_CURRENT_SOURCE_DIR}/SBarcodeFilter.h
-    )
-    
-    if(ANDROID)
-        find_package(Qt5 COMPONENTS AndroidExtras REQUIRED)
-    endif()
+set(SRC_FILES ${COMMON_SOURCES}
+    ${CMAKE_CURRENT_SOURCE_DIR}/SBarcodeScanner.cpp
+)
+set(HEADER_FILES ${COMMON_HEADERS}
+    ${CMAKE_CURRENT_SOURCE_DIR}/SBarcodeScanner.h
+)
+if(ANDROID)
+    find_package(Qt6 COMPONENTS GuiPrivate REQUIRED)
 endif()
 
 
@@ -322,10 +306,10 @@ endif()
 
 
 target_link_libraries(${PROJECT_NAME} PUBLIC
-    Qt${QT_VERSION_MAJOR}::Core
-    Qt${QT_VERSION_MAJOR}::Multimedia
-    Qt${QT_VERSION_MAJOR}::Concurrent
-    Qt${QT_VERSION_MAJOR}::Quick
+    Qt6::Core
+    Qt6::Multimedia
+    Qt6::Concurrent
+    Qt6::Quick
 )
 target_include_directories(${PROJECT_NAME} 
     PUBLIC
@@ -336,10 +320,5 @@ target_include_directories(${PROJECT_NAME}
 )
 
 if (ANDROID)
-    if(QT_VERSION_MAJOR EQUAL 5)
-        target_link_libraries(${PROJECT_NAME} PUBLIC Qt5::AndroidExtras)
-    else()
-        target_link_libraries(${PROJECT_NAME} PUBLIC Qt::GuiPrivate)
-    endif()
-
-endif()
+    target_link_libraries(${PROJECT_NAME} PUBLIC Qt6::CorePrivate)
+endif ()
diff --git a/src/zxing-cpp b/src/zxing-cpp
index a91d306..d6068bc 160000
--- a/src/zxing-cpp
+++ b/src/zxing-cpp
@@ -1 +1 @@
-Subproject commit a91d3065cc4af1ac335acf3ec1a3956ef82a4930
+Subproject commit d6068bcebeb8fd9f0d35a99b00d202be86a14dbe
